src/chat/ChatScreen.js

```diff
@@ -19,27 +23,86 @@ type Props = {
   },
 };
 
-export default class ChatScreen extends PureComponent<Props> {
+type State = {
+  editMessage: ?EditMessage,
+};
+
+class ChatScreen extends PureComponent<Props, State> {
   context: Context;
   props: Props;
+  state: State;
+
+  state = {
+    editMessage: null,
+  };
 
   static contextTypes = {
     styles: () => null,
   };
 
+  componentDidMount() {
+    BackHandler.addEventListener('hardwareBackPress', this.handleBackPress);
+  }
+
+  componentWillUnmount() {
+    BackHandler.removeEventListener('hardwareBackPress', this.handleBackPress);
+  }
+
+  handleBackPress = () => {
+    const { editMessage } = this.state;
+    if (editMessage) {
+      this.cancelEditMode();
+      // do not propagate back event to navigation
+      return true;
+    }
+    // call react-navigation back handler implementation
+    return false;
+  };
+
+  cancelEditMode = () => {
+    this.setState({
+      editMessage: null,
+    });
+  };
+
+  onEditMessageSelect = async (editMessage: Message) => {
+    const { auth } = this.props;
+    const message = await getMessageContentById(auth, editMessage.id);
+    this.setState({
+      editMessage: {
+        id: editMessage.id,
+        content: message,
+        topic: editMessage.subject,
+      },
+    });
+  };
+
   render() {
+    const { editMessage } = this.state;
     const { styles } = this.context;
     const { narrow } = this.props.navigation.state.params;
 
     return (
       <ActionSheetProvider>
         <View style={styles.screen}>
           <ZulipStatusBar narrow={narrow} />
-          <MainNavBar narrow={narrow} />
+          <MainNavBar
+            cancelEditMode={editMessage ? this.cancelEditMode : undefined}
```

Comment:
I am concerned.
When looking at this change and `Chat`, it should be concerning you too.
We should try to localize a change and to not make all components aware of it.